### PR TITLE
Support contains operator

### LINKED ISSUE #61: Adding operator
I'm looking for an operators used in other projects :

|  Operator | Description  |
|---|---|
| =~ | Matches a [JavaScript regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions). For example, [?(@.description =~ /cat.*/i)] matches items whose description starts with cat (case-insensitive).  |
|  contains | Checks if a string contains the specified substring (case-sensitive), or an array contains the specified element.  |

### PR DESCRIPTION
**Pull request by KyleJamesWalker**
### Description

Add the contains operator.

Note: This hasn't been tested yet, I wanted to see about getting some feedback before proceeding with this PR.

### Issues Resolved

#61 

### Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: jsonpath2/expressions/operator.py [hunk 1]
```diff
@@ -193,6 +193,21 @@ def __evaluate__(x_obj, y_obj):
         return False
 
 
+class InBinaryOperatorExpression(BinaryOperatorExpression):
```
Comment (kuldeepluvani) [inline jsonpath2/expressions/operator.py:196]: Does the `in` consider as BinaryOperator or just Operator?

**Review by kuldeepluvani — DISMISSED** [review]
💯

File: jsonpath2/expressions/operator.py (hunk 1 above)
Comment (KyleJamesWalker) [reply jsonpath2/expressions/operator.py:196]: It's the subclass all the other equality checks are using.

**Comment by dmlb2000** [comment]
Can you give some examples in other places where `~=` represents the `in` operation? Most of the places I'm aware of `~=` is more of a regular expression match. Not what the `in` operator does in Python.

Also looks like you need some testing to cover the new code you created.

Thoughts?

**Comment by KyleJamesWalker** [comment]
I'm currently trying to create some unit tests, it took me a little bit to get the antlr4 code generated, and now seeing what I might of missed to get past the:
```
token recognition error at: '~'
```

And let me look more into the the ~= vs contains, there's a good chance you're right, I was mainly just trying to figure out how to add to the code, before finalizing (hope it's not too distracting for you), thanks for the library!

**Comment by KyleJamesWalker** [comment]
@dmlb2000 I think this should cover the cases for contains, let me know if you'd like more tests?

Also it would be pretty easy to add the `~=` to do a regex match on the value, would you like  me to add that also?  For my use case I really only needed a substring, but I can see where a full regex would be useful.

File: docs/exampleusage.md [hunk 2]
```diff
@@ -49,6 +49,10 @@ matching.
 | ``and``, ``or``,  | Boolean operator, where operands are JSONPath |
 | ``not``           | filter expressions                            |
 +-------------------+-----------------------------------------------+
+| ``contains``      | Checks if a string contains the specified     |
```
Comment (KyleJamesWalker) [inline docs/exampleusage.md:52]: Text copied from: https://support.smartbear.com/alertsite/docs/monitors/api/endpoint/jsonpath.html

**Comment by KyleJamesWalker** [comment]
@dmlb2000 I've also started adding the `~=` but I'll have to understand the antlr4 syntax a bit more to allow escape characters, and possibly passing flags along with it.

I'll create a PR against this repo later but if you want to look at the possible changes I have a working branch on my fork if you have any advice: https://github.com/KyleJamesWalker/python-jsonpath2/pull/1

**Comment by KyleJamesWalker** [comment]
Example Error Message:
```
self = <jsonpath2.parser._ConsoleErrorListener object at 0x10746b940>
recognizer = <jsonpath2.parser.JSONPathLexer.JSONPathLexer object at 0x1070653c0>
offendingSymbol = None, line = 1, column = 24
msg = 'token recognition error at: \'"^Sayings\\s\''
e = LexerNoViableAltException(None)

    def syntaxError(self, recognizer, offendingSymbol, line, column, msg, e):
>       raise ValueError("line {}:{} {}".format(line, column, msg))
E       ValueError: line 1:24 token recognition error at: '"^Sayings\s'
```

With: `Path.parse_str('$..book[*][?(@.title ~= "^Sayings\\s.*\\sCentury$")]')`

**Comment by dmlb2000** [comment]
@KyleJamesWalker I agree the regex will be a challenge to get the regex to pass through antlr to apply the standard regex library. However, if you feel you have the ability/time to do that I can review and help a little. Feel free to submit a new bug with the regex addition and make a merge request.

**Comment by KyleJamesWalker** [comment]
@dmlb2000 How do you feel about this PR. Once this is merged, I can try and work on some more ideas on how to pass the flag to the regex matching. I at least have a small start so far with the 2nd PR.

**Review by dmlb2000 — APPROVED** [review]
Thanks for the contribution. Looks good to me!

**Pull request merged**